spec/models/article_spec.rb

```diff
@@ -630,5 +630,46 @@ def assert_sets_trigger(art)
     end
 
   end
-end
 
+  describe "#merge_with" do
+    it "returns an article" do
+      article = Factory(:article)
+      another_article = Factory(:article)
+      expect(article.merge_with(another_article.id)).to be_an_instance_of Article
+    end
+
+    it "combines the article text" do
+      article = Factory(:article, :body => "something")
+      another_article = Factory(:article, :body => "else")
+      article.merge_with(another_article.id)
+      expect(article.body).to eq "something" + "\n\n" + "else"
+    end
+
+    it "has the title of one of the articles" do
+      article = Factory(:article, :title => "something")
+      another_article = Factory(:article, :title => "else")
+      article.merge_with(another_article.id)
+      expect(article.title).to eq "something"
+    end
+
+    it "carries over the comments from both articles" do
+      article = Factory(:article)
+      comment = Factory(:comment, :article_id => article.id)
+      original_comments_count = article.comments.count
+      another_article = Factory(:article)
+      another_comment = Factory(:comment, :article_id => another_article.id)
+      article.merge_with(another_article.id)
```

Comment:
This is a pretty big test with setup and expectations, maybe just separate the two with some whitespace for readability